packages/amplify-codegen/src/commands/generateStatementsAndType.js

```diff
@@ -17,41 +17,42 @@ async function generateStatementsAndTypes(context, forceDownloadSchema, maxDepth
     withoutInit = true;
   }
 
-  // Check if introspection schema exists
-  const schemaPath = ['schema.graphql', 'schema.json'].map(p => path.join(process.cwd(), p)).find(p => fs.existsSync(p));
-  if (withoutInit && !schemaPath) {
-    throw Error(
-      `Please download the schema.graphql or schema.json and place in ${process.cwd()} before adding codegen when not in an amplify project`
-    );
-  }
-
-  if (withoutInit) {
-    forceDownloadSchema = false;
-  }
   const config = loadConfig(context, withoutInit);
   const projects = config.getProjects();
   if (!projects.length) {
     throw new NoAppSyncAPIAvailableError(constants.ERROR_CODEGEN_NO_API_CONFIGURED);
   }
+  const project = projects[0];
+  const schemaPath = ['schema.graphql', 'schema.json'].map(p => path.join(process.cwd(), p)).find(p => fs.existsSync(p));
+  if (withoutInit && !project && !schemaPath) {
+    throw Error(
+      `Please download the schema.graphql or schema.json and place in ${process.cwd()} before adding codegen when not in an amplify project`,
+    );
+  }
+
   let apis = [];
   if (!withoutInit) {
     apis = getAppSyncAPIDetails(context);
+  } else if (project.amplifyExtension.apiId && project.amplifyExtension.region) {
+    const {
+      amplifyExtension: { apiId, region },
+    } = project;
+    apis = [await getAppSyncAPIInfo(context, apiId, region)];
```

Comment:
You could probably refactor this check, destructuring statement, and retrieval into a single utility where given a `project` you return an optional `api` or something like that, since it's nearly identical to the code in `statements.ts`